src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -426,7 +426,15 @@ void refreshForeground()
 			if (Foreground?.SupportsAssignAndObserveBrush ?? false)
 			{
 				_foregroundChanged.Disposable =
+#if __SKIA__
+					Brush.AssignAndObserveBrush(Foreground, Visual.Compositor, compositionBrush =>
+					{
+						refreshForeground();
+						ForegroundCompositionBrush = compositionBrush;
+					});
+#else
 					Brush.AssignAndObserveBrush(Foreground, c => refreshForeground(), refreshForeground);
+#endif
```

Comment:
That's annoying to have platform-specific code there.

Reply:
@carldebilly I agree it's annoying. But the only other approach I could think of is introducing a partial, which will duplicate the non-Skia code for Android, macOS, iOS, and Wasm. Do you have other ideas?